pkg/ui/chooser/routes.go

```diff
@@ -8,23 +8,52 @@ import (
 	"github.com/gofiber/fiber/v2"
 	"github.com/gofiber/fiber/v2/middleware/etag"
 	"github.com/gofiber/fiber/v2/middleware/filesystem"
+	"github.com/gofiber/template/html/v2"
 
+	"github.com/G-Research/fasttrackml/pkg/ui/chooser/controller"
 	"github.com/G-Research/fasttrackml/pkg/ui/common"
 )
 
 //go:embed embed
 var content embed.FS
 
-func AddRoutes(r fiber.Router) {
-	sub, _ := fs.Sub(content, "embed")
+// Router represents `chooser` router.
+type Router struct {
+	controller *controller.Controller
+}
+
+// NewRouter creates new instance of `chooser` router.
+func NewRouter(controller *controller.Controller) *Router {
+	return &Router{
+		controller: controller,
+	}
+}
 
-	r.Use("/static/chooser/", etag.New(), filesystem.New(filesystem.Config{
+// AddRoutes adds all the `chooser` routes
+func (r Router) AddRoutes(fr fiber.Router) error {
+	sub, err := fs.Sub(content, "embed")
+	if err != nil {
+		return err
+	}
+
+	fr.Use("/static/chooser/", etag.New(), filesystem.New(filesystem.Config{
 		Root: http.FS(sub),
 	}))
 
-	r.Use("/", etag.New(), filesystem.New(filesystem.Config{
+	// app for template rendering
+	app := fiber.New(fiber.Config{
+		Views: html.NewFileSystem(http.FS(sub), ".html"),
+	})
+	fr.Mount("/", app)
+
+	// specific routes
+	app.Get("/", r.controller.GetNamespaces)
+
+	// default route
+	app.Use("/", etag.New(), filesystem.New(filesystem.Config{
 		Root: http.FS(
 			common.NewOnlyRootFS(sub, "index.html"),
 		),
 	}))
+	return nil
```

Comment:
there is an overlap between these various routes and the last one does not get used